examples/renode/freertos_multitask/main.cpp

```
@@ -0,0 +1,165 @@
+};
+
+static QueueHandle_t msg_queue;
+
+static void producer_task(void *) {
+    printf("\n--- Producer: starting ---\n");
+
+    for (int i = 0; i < NUM_MESSAGES; i++) {
+        auto msg = someip::platform::allocate_message();
+        if (!msg) {
+            printf("  Producer: allocation failed at message %d\n", i);
+            continue;
```

Cppcheck (2.20.0)

[style] The function 'get_message_id' is never used.

(unusedFunction)